packages/history/src/index.ts

```diff
@@ -174,20 +174,22 @@ function assignKey(state: HistoryState) {
  * @param opts
  * @param opts.getHref A function that returns the current href (path + search + hash)
  * @param opts.createHref A function that takes a path and returns a href (path + search + hash)
+ * @param opts.win A reference to the Window object
  * @returns A history instance
  */
 export function createBrowserHistory(opts?: {
   getHref?: () => string
   createHref?: (path: string) => string
+  win: Window = window
 }): RouterHistory {
   const getHref =
     opts?.getHref ??
     (() =>
-      `${window.location.pathname}${window.location.search}${window.location.hash}`)
+      `${win.location.pathname}${win.location.search}${win.location.hash}`)
```

Comment:
Perhaps rename the `win` to `_window` or to be consistent, refactor it to a callback instead.

getWindow()?: () => Window

Reply:
What would be the benefit of a callback? It would only be called once as we otherwise would need some sort of subscription to get informed about "possible" changes of the returned window instance.

I still updated the signature slightly, as there was an error inside